### PR TITLE
Add seamless upgrade for OpenShift installations (1.2.2 to 1.3.3)

### PR DESCRIPTION
**Pull request by nikhil-thomas**
Add a kubernetes job to remove `config.operator.tekton.dev` CRD on successful
installation of Tekton components on OpenShift (to trigger deletions
of obsolete resources, which has `config` instance as ownerReference).

Modify Tekton-Config reconciler OpenShift extension postInstall logic to
run the Job at the appropriate time.

Remove clustertasks v0.16 which were not removed during the payload update

**Note:** the upgrade process deletes all old version-named clustertasks as their ownerReference will not be updated in the new install. I shall make a separate patch to handle this.

Signed-off-by: Nikhil Thomas <nikthoma@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: pkg/reconciler/openshift/tektonconfig/extension/openshift_cleanup.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	status := instance.GetStatus()
+	if err := manifest.Apply(); err != nil {
+		status.MarkInstallFailed(err.Error())
+		return fmt.Errorf("failed to apply cleanup job: %w", err)
+	}
+	return nil
+}
+
+func CheckCleanup(ctx context.Context, manifest *mf.Manifest, instance v1alpha1.TektonComponent) error {
+	return nil
```
Comment (savitaashture) [inline pkg/reconciler/openshift/tektonconfig/extension/openshift_cleanup.go:41]: As its returning nil do we require this function ?

Comment (nikhil-thomas) [reply pkg/reconciler/openshift/tektonconfig/extension/openshift_cleanup.go:41]: i kepth it as a place holder to implement the logic to check whether the job was successful.
I should have added a  `TODO:` comment.

Shall I handle it in my next pr.

**Review by vdemeester — APPROVED** [review]

File: pkg/reconciler/openshift/tektonconfig/extension/openshift_cleanup.go (hunk 1 above)
Comment (savitaashture) [reply pkg/reconciler/openshift/tektonconfig/extension/openshift_cleanup.go:41]: Sure :+1:

**Pull request merged**